src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -389,24 +393,37 @@ RouterModifyTaskForShardInterval(Query *originalQuery, ShardInterval *shardInter
 						errdetail("Select query cannot be pushed down to the worker.")));
 	}
 
-	/* Ensure that we have INSERTed table's placement exists on the same worker */
-	insertShardPlacementList = FinalizedShardPlacementList(shardId);
-	intersectedPlacementList = IntersectPlacementList(insertShardPlacementList,
-													  selectPlacementList);
 
-	if (list_length(insertShardPlacementList) != list_length(intersectedPlacementList))
+	/* ensure that we do not send queries where select is pruned away completely */
+	if (list_length(selectPlacementList) == 0)
 	{
 		ereport(DEBUG2, (errmsg("could not generate task for target shardId: %ld",
```

Comment:
Hmm, I think we actually could not generate task and the error detail already mentions that. 
Should I reword the errmsg as well?